pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/IndexCreationContext.java

```diff
@@ -0,0 +1,454 @@
+    public int getMaxNumberOfMultiValueElements() {
+      return _delegate.getMaxNumberOfMultiValueElements();
+    }
+
+    @Override
+    public int getMaxRowLengthInBytes() {
+      return _delegate.getMaxRowLengthInBytes();
+    }
+
+    @Override
+    public boolean isSorted() {
+      return _delegate.isSorted();
+    }
+
+    @Override
+    public int getCardinality() {
+      return _delegate.getCardinality();
+    }
+
+    @Override
+    public int getTotalNumberOfEntries() {
+      return _delegate.getTotalNumberOfEntries();
+    }
+
+    @Override
+    public int getTotalDocs() {
+      return _delegate.getTotalDocs();
+    }
+
+    @Override
+    public boolean hasDictionary() {
+      return _delegate.hasDictionary();
+    }
+  }
+
+  class BloomFilter extends Wrapper {
+
+    private final BloomFilterConfig _bloomFilterConfig;
+
+    public BloomFilter(IndexCreationContext wrapped, BloomFilterConfig bloomFilterConfig) {
+      super(wrapped);
+      _bloomFilterConfig = bloomFilterConfig;
+    }
+
+    @Nullable
+    public BloomFilterConfig getBloomFilterConfig() {
+      return _bloomFilterConfig;
+    }
+  }
+
+  class Forward extends Wrapper {
+
+    private final ChunkCompressionType _chunkCompressionType;
+    private final Map<String, Map<String, String>> _columnProperties;
+
+    Forward(IndexCreationContext delegate,
+        ChunkCompressionType chunkCompressionType,
+        @Nullable Map<String, Map<String, String>> columnProperties) {
+      super(delegate);
+      _chunkCompressionType = chunkCompressionType;
+      _columnProperties = columnProperties;
+    }
+
+    public ChunkCompressionType getChunkCompressionType() {
+      return _chunkCompressionType;
+    }
+
+    @Nullable
+    public Map<String, Map<String, String>> getColumnProperties() {
+      return _columnProperties;
+    }
+  }
+
+  class Geospatial extends Wrapper {
+
+    private final H3IndexConfig _h3IndexConfig;
+
+    Geospatial(IndexCreationContext delegate, H3IndexConfig h3IndexConfig) {
+      super(delegate);
+      _h3IndexConfig = h3IndexConfig;
+    }
+
+    public H3IndexConfig getH3IndexConfig() {
+      return _h3IndexConfig;
+    }
+  }
+
+  class Inverted extends Wrapper {
+
+    Inverted(IndexCreationContext delegate) {
+      super(delegate);
+    }
+  }
+
+  class Json extends Wrapper {
+
+    Json(IndexCreationContext delegate) {
+      super(delegate);
+    }
+  }
+
+  class Range extends Wrapper {
+
+    private final ColumnMetadata _columnMetadata;
+    private final int _rangeIndexVersion;
+
+    Range(IndexCreationContext delegate, ColumnMetadata columnMetadata, int rangeIndexVersion) {
+      super(delegate);
+      _columnMetadata = columnMetadata;
+      _rangeIndexVersion = rangeIndexVersion;
+    }
+
+    public ColumnMetadata getColumnMetadata() {
+      return _columnMetadata;
+    }
+
+    public int getRangeIndexVersion() {
+      return _rangeIndexVersion;
+    }
+  }
+
+  class Text extends Wrapper {
```

Comment:
(nit) would be good to add a short note on 2 constructors highlighting the first one is for Lucene text and other one is for 2 flavors of FST.

Reply:
Done